### PR TITLE
Move stages to each algorithms

### PR DESCRIPTION
**Pull request by jaegukhyun**
### Summary
This PR is for moving stages in mpa to each algorithms
otx.mpa.cls ==> otx.algorithms.classification.adapters.mmcls.tasks(new folder)
otx.mpa.det ==> otx.algorithms.detection.adapters.mmdet.tasks(new folder)
otx.mpa.seg ==> otx.algorithms.segmentation.adapters.mmseg.tasks(new folder)

### Note.
These stages(trainer, inferrer, etc) will be replaced in Phase2 refactoring(removing callstack)

### How to test
Normal intg, e2e, unit test

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md)
- [x] I have updated the [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) accordingly
- [x] I have added tests to cover my changes
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2023 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: otx/algorithms/segmentation/adapters/mmseg/tasks/incremental/stage.py [hunk 1]
```diff
@@ -9,26 +10,28 @@
...
+    def configure_task(self, cfg, training):
+        """Adjust settings for task adaptation."""
+        super().configure_task(cfg, training)
 
         new_classes = np.setdiff1d(self.model_classes, self.org_model_classes).tolist()
 
         # FIXME : can be naive supervised learning (from-scratch ver.)
         # Check if new classes are added
-        has_new_class = True if len(new_classes) > 0 else False
+        has_new_class = len(new_classes)
```
Comment (eunwoosh) [inline otx/algorithms/segmentation/adapters/mmseg/tasks/incremental/stage.py]:
```suggestion
        has_new_class = len(new_classes) > 0
```
0 is False == False

File: otx/algorithms/detection/adapters/mmdet/tasks/trainer.py [hunk 2]
```diff
@@ -103,7 +107,7 @@ def run(self, model_cfg, model_ckpt, data_cfg, **kwargs):
         # cfg.dump(osp.join(cfg.work_dir, 'config.py'))
         # logger.info(f'Config:\n{cfg.pretty_text}')
 
-        validate = True if cfg.data.get("val", None) else False
+        validate = cfg.data.get("val", None)
```
Comment (eunwoosh) [inline otx/algorithms/detection/adapters/mmdet/tasks/trainer.py]:
```suggestion
        validate = "val" in cfg.data
```
validate should be boolean type

File: otx/algorithms/common/adapters/mmcv/tasks/stage.py [hunk 3]
```diff
@@ -499,19 +512,20 @@ def read_label_schema(ckpt_path, name_only=True, file_name="label_schema.json"):
...
-        # InferenceProgressCallback (Time Monitor enable into Infer task)
+        """Inferenceprogresscallback (Time Monitor enable into Infer task)."""
         time_monitor = None
         if cfg.get("custom_hooks", None):
             time_monitor = [hook.time_monitor for hook in cfg.custom_hooks if hook.type == "OTXProgressHook"]
             time_monitor = time_monitor[0] if time_monitor else None
         if time_monitor is not None:
 
-            def pre_hook(module, input):
+            def pre_hook(*args):
```
Comment (eunwoosh) [inline otx/algorithms/common/adapters/mmcv/tasks/stage.py]:
```suggestion
            def pre_hook(*args, **kwargs):
```
Isn't kwargs needed?

File: otx/algorithms/common/adapters/mmcv/tasks/stage.py [hunk 4]
```diff
@@ -499,19 +512,20 @@ def read_label_schema(ckpt_path, name_only=True, file_name="label_schema.json"):
...
                 time_monitor.on_test_batch_begin(None, None)
 
-            def hook(module, input, output):
+            def hook(*args):
```
Comment (eunwoosh) [inline otx/algorithms/common/adapters/mmcv/tasks/stage.py]:
```suggestion
            def hook(*args, **kwargs):
```
Isn't kwargs needed?

File: otx/algorithms/segmentation/adapters/mmseg/tasks/trainer.py [hunk 5]
```diff
@@ -96,7 +99,7 @@ def run(self, model_cfg, model_ckpt, data_cfg, **kwargs):
         # cfg.dump(osp.join(cfg.work_dir, 'config.py'))
         # logger.info(f'Config:\n{cfg.pretty_text}')
 
-        validate = True if cfg.data.get("val", None) else False
+        validate = cfg.data.get("val", None)
```
Comment (eunwoosh) [inline otx/algorithms/segmentation/adapters/mmseg/tasks/trainer.py]:
```suggestion
        validate = "val" in cfg.data
```

**Review by harimkang — DISMISSED** [review]

**Review by supersoob — DISMISSED** [review]

**Review by eunwoosh — DISMISSED** [review]
Thanks for your work!

**Review by harimkang — DISMISSED** [review]

**Review by eunwoosh — DISMISSED** [review]

**Review by eunwoosh — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

**Review by supersoob — APPROVED** [review]

**Pull request merged**